morebits.js

```diff
@@ -1402,22 +1402,21 @@ Morebits.unbinder.getCallback = function UnbinderGetCallback(self) {
  */
 Morebits.date = function() {
 	var args = Array.prototype.slice.call(arguments);
+	this._d = new (Function.prototype.bind.apply(Date, [Date].concat(args)));
 
-	// Date.parse implementations vary too much between browsers, and
-	// MediaWiki's format is too non-standard, so we just convert MW
-	// timestamps to ISO-8601. A paren-wrapped 'UTC' messes everyone up,
-	// and the comma after the time is only okay in modern Firefox. After
-	// this first replace, Chrome and Firefox are content. The second
-	// replace is mainly for Safari, which basically *only* accepts the
-	// simplified ECMA-262 implementation of ISO-8601.
-	if (typeof args[0] === 'string') {
-		args[0] = args[0].replace(/(\d\d:\d\d),/, '$1').replace(/\(UTC\)/, 'UTC');
-		// Safari is particular about timezone offsets, so this is intentionally specific
-		args[0] = args[0].replace(/(\d\d:\d\d) (\d{1,2}) ([A-Z][a-z]+) (\d{4}) UTC$/, function(match, time, date, monthname, year) {
-			return [year, mw.config.get('wgMonthNames').indexOf(monthname), date].join('-') + 'T' + time + 'Z';
-		});
+	if (!this.isValid()) {
+		if (args.length === 1 && typeof args[0] === 'string') {
+			// Check if it's a MediaWiki signature timestamp (which the native Date cannot parse directly)
+			var dateParts = Morebits.date.localeData.signatureTimestampFormat(args[0]);
+			if (dateParts) {
+				this._d = new Date(Date.UTC.apply(null, dateParts));
+			}
+		}
+	}
+	// Still no?
+	if (!this.isValid()) {
+		mw.log.warn('Invalid Morebits.date initialisation:', args);
```

Comment:
Unless you've any opposition, I can just take care of this and merge this.

Reply:
Doesn't matter either way, though I thought we were already using these for those deprecation warnings. I don't see this as a rule violation since the rule really is to avoid debug logging which this isn't, but then I'm not a rules-oriented guy!

Reply:
100% agreed.  I'm fine with converting all actual errors and deprecation notices to `mw.log`, I only meant that this would imply/precipitate that move without actually discussing it!